### PR TITLE
added 'Aborting' and 'Aborted' submission statuses

### PR DESCRIPTION
**Pull request by MatthewBemis**
Ready for review, but has not yet been tested outside of unit tests.

https://broadinstitute.atlassian.net/browse/DSDEEPB-2469
- [x] **Submitter**: Rebase to develop. DO NOT SQUASH
- [x] **Submitter**: Make sure Swagger is updated if API changes
- [x] **Submitter**: Make sure documentation for code is complete
- [x] **Submitter**: Review code comments; remove done TODOs, create stories for remaining TODOs
- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Add description or comments on the PR explaining the hows/whys (if not obvious)
- [x] Tell  that the PR exists if he wants to look at it
- [x] Anoint a lead reviewer (LR). **Assign PR to LR**
- [x] **LR**: Initial review by LR and others.
- [x] Comment / review / update cycle:
  - Rest of team may comments on PR at will
  - **LR assigns to submitter** for feedback fixes
  - Submitter updates documentation as needed
  - Submitter rebases to develop again if necessary
  - Submitter makes further commits. DO NOT SQUASH. **Reassign to LR** for further feedback
- [x]  sign off
- [x] **LR** sign off
- [ ] **Assign to submitter** to finalize
- [ ] **Submitter**: Squash commits, rebase if necessary
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Merge to develop 
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: Check configuration files in Jenkins in case they need changes
- [ ] **Submitter**: Verify swagger UI on dev server still works after deployment
- [ ] **Submitter**: Inform other teams of any API changes via hipchat and/or email
- [ ] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package org.broadinstitute.dsde.rawls.model
 
 import org.broadinstitute.dsde.rawls.model.SubmissionStatuses.SubmissionStatus
-import org.broadinstitute.dsde.rawls.model.WorkflowStatuses.WorkflowStatus
+import org.broadinstitute.dsde.rawls.model.WorkflowStatuses.{Aborted, Aborting, WorkflowStatus}
```
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/rawls/model/ExecutionModel.scala]: this change is unnecessary

File: src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala [hunk 2]
```diff
@@ -147,7 +147,13 @@ class SubmissionMonitor(workspaceName: WorkspaceName,
...
 
         if (refreshedSubmission.workflows.forall(workflow => workflow.status.isDone)) {
-          containerDAO.submissionDAO.update(workspaceContext, refreshedSubmission.copy(status = SubmissionStatuses.Done), txn)
+          val originalStatus = containerDAO.submissionDAO.get(workspaceContext, refreshedSubmission.submissionId, txn).get.status
+          originalStatus match {
+            case SubmissionStatuses.Submitted =>
+              containerDAO.submissionDAO.update(workspaceContext, refreshedSubmission.copy(status = SubmissionStatuses.Done), txn)
+            case SubmissionStatuses.Aborting =>
+              containerDAO.submissionDAO.update(workspaceContext, refreshedSubmission.copy(status = SubmissionStatuses.Aborted), txn)
+          }
```
Comment (helgridly) [inline src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala]: is this going to throw a match error if the original isn't one of these two?

**Comment by helgridly** [comment]
:+1: but bear in mind that Cromwell can take a _long_ time to move out of Aborting into a terminal state, which will be horribly confusing to users if they expand down to workflows. They might even see something like Aborting/Running for a while.

File: src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala (hunk 2 above)
Comment (MatthewBemis) [inline src/main/scala/org/broadinstitute/dsde/rawls/jobexec/SubmissionMonitor.scala]: It should only ever be Submitted or Aborting in this stage, but it probably should be more robust than this. Will address.

**Comment by dvoet** [comment]
:+1:

**Pull request merged**